src/js/components/sharedComponents/autocomplete/Autocomplete.jsx

```diff
@@ -0,0 +1,242 @@
+/**
+ * Created by michaelbray on 1/27/17.
+ */
+
+import React from 'react';
+import _ from 'lodash';
+
+import Warning from './Warning';
+import SuggestionHolder from './SuggestionHolder';
+
+const propTypes = {
+    values: React.PropTypes.array,
+    placeholder: React.PropTypes.string,
+    handleTextInput: React.PropTypes.func,
+    onSelect: React.PropTypes.func,
+    errorHeader: React.PropTypes.string,
+    errorMessage: React.PropTypes.string,
+    tabIndex: React.PropTypes.number,
+    isRequired: React.PropTypes.bool
+};
+
+const defaultProps = {
+    values: [],
+    placeholder: '',
+    tabIndex: null,
+    isRequired: false,
+    errorHeader: null,
+    errorDescription: null
+};
+
+export default class Autocomplete extends React.Component {
+    constructor(props) {
+        super(props);
+
+        this.dataDictionary = {};
+
+        this.state = {
+            value: '',
+            shown: '',
+            selectedIndex: 0,
+            showWarning: ''
+        };
+    }
+
+    componentDidMount() {
+        this.setupAutocomplete();
+    }
+
+    componentDidUpdate(prevProps) {
+        if (!_.isEqual(prevProps.values, this.props.values)) {
+            this.open();
+        }
+    }
+
+    onChange(e) {
+        this.checkValidity(e.target.value);
+        this.props.handleTextInput(e);
```

Comment:
Should `state.selectedIndex` reset to zero when the autocomplete results change? It currently doesn't, so if you have second item highlighted and then change the search term, the second item of the new list is pre-selected.